code/modules/mob/living/simple_animal/friendly/iriska.dm

```diff
@@ -0,0 +1,203 @@
+/mob/living/simple_animal/iriska
+	name = "Iriska"
+	desc = "The captain's own cat. Fat and lazy."
+	icon = 'icons/mob/iriska.dmi'
+	icon_state = "iriska"
+	health = 80
+	maxHealth = 80
+	wander = FALSE
+	canmove = FALSE
+	speak_emote = list("purrs.", "meows.")
+	emote_see = list("shakes her head.", "shivers.")
+	speak_chance = 0.75
+	meat_amount = 6
+	meat_type = /obj/item/reagent_containers/food/snacks/meat
+	response_help = "pets"
+	response_disarm = "rubs"
+	response_harm = "makes terrible mistake by kicking"
+	min_oxy = 16
+	minbodytemp = 223
+	maxbodytemp = 323
+	mob_size = MOB_HUGE
+	harm_intent_damage = 20
+	melee_damage_lower = 10
+	melee_damage_upper = 30
+	attacktext = "slashed"
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+	density = TRUE
+	bite_factor = 0.8
+	stomach_size_mult = 10
+
+	autoseek_food = FALSE // Original code for seeking food doesn't fit, Iriska will use slightly modified version
+	beg_for_food = FALSE
+	max_scan_interval = 10
+	eat_from_hand = FALSE
+
+var/atom/snack = null
+
+var/befriend_job = "Captain"
+
+var/list/mob/living/carbon/human/tolerated = list()
```

Comment:
You want to avoid keeping direct references to things, as it prevents them being properly deleted.

Reply:
This means that if a mob on this list is deleted (by being gibed as an example), due to the fact that it's on this list it won't be able to be deleted properly. You need to either not hold a direct reference to the mob (like checking DNA or some other identifier) or remove the mob from this list on its destroy proc (mob would need to reference the cat so it knows what list to remove itself from, would get complicated)